src/provider.toit

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2023 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file.
+
+import esp32
+import log
+import monitor
+import system.services show ServiceProvider ServiceResource ServiceHandler
+import .service
+
+class WatchdogServiceProvider extends ServiceProvider
+  implements ServiceHandler:
+
+  static GRANULARITY-MS ::= 2_000
+
+  dogs_/Map ::= {:}  // From string to Watchdog.
+  system-watchdog-task_/Task? := null
+  logger_/log.Logger := (log.default.with-name "watchdog").with-level log.INFO-LEVEL
+  mutex_/monitor.Mutex := monitor.Mutex
+
+  constructor:
+    super "watchdog" --major=1 --minor=0
```

Comment:
The API is 0.1, but the implementation is 1.0?

Reply:
We have done the same in our service tutorial.
Probably should clean it up there too. Going with v1.0 everywhere now.